### ISSUE TITLE
Typed static property verbb\hyper\Hyper::$plugin must not be accessed before initialization

### ISSUE THREAD

**Issue by mgambella** [verbb/hyper#83]
### Describe the bug

After installing GraphQL Authentication 2.5.0 and Hyper 1.1.10 and creating an Hyper field I get an error across the entire dashboard rendering Craft CMS completely unaccessible.

Here is the error present in the logs files:

```
2023-08-04 13:04:44 [web.INFO] [yii\db\Connection::open] Opening DB connection: mysql:host=db;dbname=db;port=3306 {"memory":1854504} 
2023-08-04 13:04:44 [web.INFO] [yii\web\Session::open] Session started {"memory":3667184} 
2023-08-04 13:04:44 [web.INFO] [nystudio107\codeeditor\CodeEditor::bootstrap] CodeEditor module bootstrapped {"memory":3901360} 
2023-08-04 13:04:44 [web.WARNING] [craft\elements\db\ElementQuery::prepare] Element query executed before Craft is fully initialized.
Stack trace:
#0 /var/www/html/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare()
#1 /var/www/html/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build()
#2 /var/www/html/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Query->createCommand()
#3 /var/www/html/vendor/craftcms/cms/src/db/Query.php(275): yii\db\Query->one()
#4 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1590): craft\db\Query->one()
#5 /var/www/html/vendor/craftcms/cms/src/elements/User.php(525): craft\elements\db\ElementQuery->one()
#6 /var/www/html/vendor/yiisoft/yii2/web/User.php(698): craft\elements\User::findIdentity()
#7 /var/www/html/vendor/craftcms/cms/src/web/User.php(500): yii\web\User->renewAuthStatus()
#8 /var/www/html/vendor/yiisoft/yii2/web/User.php(199): craft\web\User->renewAuthStatus()
#9 /var/www/html/vendor/craftcms/cms/src/web/User.php(257): yii\web\User->getIdentity()
#10 /var/www/html/vendor/jamesedmonston/graphql-authentication/src/GraphqlAuthentication.php(207): craft\web\User->getIsAdmin()
#11 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): jamesedmonston\graphqlauthentication\GraphqlAuthentication->init()
#12 /var/www/html/vendor/yiisoft/yii2/base/Module.php(161): yii\base\BaseObject->__construct()
#13 /var/www/html/vendor/craftcms/cms/src/base/Plugin.php(122): yii\base\Module->__construct()
#14 craft\base\Plugin->__construct()
#15 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#16 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#17 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#18 /var/www/html/vendor/craftcms/cms/src/Craft.php(54): yii\BaseYii::createObject()
#19 /var/www/html/vendor/craftcms/cms/src/services/Plugins.php(943): Craft::createObject()
#20 /var/www/html/vendor/craftcms/cms/src/services/Plugins.php(228): craft\services\Plugins->createPlugin()
#21 /var/www/html/vendor/craftcms/cms/src/base/ApplicationTrait.php(1580): craft\services\Plugins->loadPlugins()
#22 /var/www/html/vendor/craftcms/cms/src/web/Application.php(108): craft\web\Application->_postInit()
#23 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#24 /var/www/html/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct()
#25 yii\base\Application->__construct()
#26 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#27 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#28 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#29 /var/www/html/vendor/craftcms/cms/src/Craft.php(54): yii\BaseYii::createObject()
#30 /var/www/html/vendor/craftcms/cms/bootstrap/bootstrap.php(256): Craft::createObject()
#31 /var/www/html/vendor/craftcms/cms/bootstrap/web.php(40): require()
#32 /var/www/html/web/index.php(11): require() {"memory":6115744} 
2023-08-04 13:04:44 [web.ERROR] [Error] Error: Typed static property verbb\hyper\Hyper::$plugin must not be accessed before initialization in /var/www/html/vendor/verbb/hyper/src/fields/HyperField.php:399
Stack trace:
#0 /var/www/html/vendor/yiisoft/yii2/base/Component.php(180): verbb\hyper\fields\HyperField->setLinkTypes(Array)
#1 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(558): yii\base\Component->__set('linkTypes', Array)
#2 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(107): yii\BaseYii::configure(Object(verbb\hyper\fields\HyperField), Array)
#3 /var/www/html/vendor/craftcms/cms/src/base/Model.php(78): yii\base\BaseObject->__construct(Array)
#4 /var/www/html/vendor/verbb/hyper/src/fields/HyperField.php(75): craft\base\Model->__construct(Array)
#5 [internal function]: verbb\hyper\fields\HyperField->__construct(Array)
#6 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs(Array)
#7 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build('verbb\\hyper\\fie...', Array, Array)
#8 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('verbb\\hyper\\fie...', Array, Array)
#9 /var/www/html/vendor/craftcms/cms/src/Craft.php(54): yii\BaseYii::createObject(Array, Array)
#10 /var/www/html/vendor/craftcms/cms/src/helpers/Component.php(143): Craft::createObject(Array)
#11 /var/www/html/vendor/craftcms/cms/src/services/Fields.php(593): craft\helpers\Component::createComponent(Array, 'craft\\base\\Fiel...')
#12 /var/www/html/vendor/craftcms/cms/src/services/Fields.php(618): craft\services\Fields->createField(Array)
#13 /var/www/html/vendor/craftcms/cms/src/services/Fields.php(647): craft\services\Fields->_fields(NULL)
#14 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(2086): craft\services\Fields->getAllFields()
#15 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1393): craft\elements\db\ElementQuery->customFields()
#16 /var/www/html/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare(Object(craft\db\mysql\QueryBuilder))
#17 /var/www/html/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build(Object(craft\elements\db\UserQuery))
#18 /var/www/html/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Query->createCommand(Object(craft\db\Connection))
#19 /var/www/html/vendor/craftcms/cms/src/db/Query.php(275): yii\db\Query->one(NULL)
#20 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1590): craft\db\Query->one(NULL)
#21 /var/www/html/vendor/craftcms/cms/src/elements/User.php(525): craft\elements\db\ElementQuery->one()
#22 /var/www/html/vendor/yiisoft/yii2/web/User.php(698): craft\elements\User::findIdentity(1)
#23 /var/www/html/vendor/craftcms/cms/src/web/User.php(500): yii\web\User->renewAuthStatus()
#24 /var/www/html/vendor/yiisoft/yii2/web/User.php(199): craft\web\User->renewAuthStatus()
#25 /var/www/html/vendor/craftcms/cms/src/web/User.php(257): yii\web\User->getIdentity()
#26 /var/www/html/vendor/jamesedmonston/graphql-authentication/src/GraphqlAuthentication.php(207): craft\web\User->getIsAdmin()
#27 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): jamesedmonston\graphqlauthentication\GraphqlAuthentication->init()
#28 /var/www/html/vendor/yiisoft/yii2/base/Module.php(161): yii\base\BaseObject->__construct(Array)
#29 /var/www/html/vendor/craftcms/cms/src/base/Plugin.php(122): yii\base\Module->__construct('graphql-authent...', Object(craft\web\Application), Array)
#30 [internal function]: craft\base\Plugin->__construct('graphql-authent...', Object(craft\web\Application), Array)
#31 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs(Array)
#32 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build('jamesedmonston\\...', Array, Array)
#33 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('jamesedmonston\\...', Array, Array)
#34 /var/www/html/vendor/craftcms/cms/src/Craft.php(54): yii\BaseYii::createObject(Array, Array)
#35 /var/www/html/vendor/craftcms/cms/src/services/Plugins.php(943): Craft::createObject(Array, Array)
#36 /var/www/html/vendor/craftcms/cms/src/services/Plugins.php(228): craft\services\Plugins->createPlugin('graphql-authent...', Array)
#37 /var/www/html/vendor/craftcms/cms/src/base/ApplicationTrait.php(1580): craft\services\Plugins->loadPlugins()
#38 /var/www/html/vendor/craftcms/cms/src/web/Application.php(108): craft\web\Application->_postInit()
#39 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#40 /var/www/html/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct(Array)
#41 [internal function]: yii\base\Application->__construct(Array)
#42 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs(Array)
#43 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build('craft\\web\\Appli...', Array, Array)
#44 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('craft\\web\\Appli...', Array, Array)
#45 /var/www/html/vendor/craftcms/cms/src/Craft.php(54): yii\BaseYii::createObject(Array, Array)
#46 /var/www/html/vendor/craftcms/cms/bootstrap/bootstrap.php(256): Craft::createObject(Array)
#47 /var/www/html/vendor/craftcms/cms/bootstrap/web.php(40): require('/var/www/html/v...')
#48 /var/www/html/web/index.php(11): require('/var/www/html/v...')
#49 {main} {"memory":6501896,"exception":"[object] (Error(code: 0): Typed static property verbb\\hyper\\Hyper::$plugin must not be accessed before initialization at /var/www/html/vendor/verbb/hyper/src/fields/HyperField.php:399)"} 
2023-08-04 13:04:44 [web.INFO] [application] Request context:
```

### Steps to reproduce

1. Install Hyper plugin 1.1.10
2. Install GraphQL Authentication plugin 2.5.0
3. Create a field with Hyper type

### Craft CMS version

4.4.16.1

### Plugin version

1.1.10

### Multi-site?

No

### Additional context

The Craft CMS installation is using also Craft Commerce and it has an headless setup, and for this reason we are also using GraphQL Authentication plugin http://plugins.craftcms.com/graphql-authentication ( https://github.com/jamesedmonston/graphql-authentication )

Disabling the GraphQL Authentication plugin solves the issue.

Can Hyper plugin be fixed so that it works with GraphQL Authentication plugin?

**Comment by engram-design** [comment]
Fixed for the next release. To get this early run `composer require verbb/hyper:"dev-craft-4 as 1.1.10"`

**Comment by engram-design** [comment]
Fixed in [1.1.11](https://github.com/verbb/hyper/releases/tag/1.1.11)
